Collision in imports of same name for different packages

The following program `sample.go` triggers a panic:

```go
package main
  
import "./p1"

func main() { println("num:", p1.Uint32())}
```
`./p1/s1.go`:
```go
package p1
  
import "crypto/rand"

var Prime = rand.Prime
```

`./p1/s2.go`: 
```go
package p1
  
import "math/rand"

var Uint32 = rand.Uint32
```

Expected result:
```console
$ go run ./sample.go
num: 2596996162
```

Got:
```console
$ yaegi ./sample.go
scope.go:163: nil reflect type
panic: nil reflect type

goroutine 1 [running]:
log.Panic(0xc0001eacd8, 0x1, 0x1)
	/usr/local/Cellar/go/1.12.1/libexec/src/log/log.go:333 +0xac
github.com/containous/yaegi/interp.(*Scope).add(0xc0000a2300, 0xc00015fee0, 0xc0001744e0)
	/tmp/src/github.com/containous/yaegi/interp/scope.go:163 +0x248
github.com/containous/yaegi/interp.(*Interpreter).Cfg.func2(0xc000220780)
	/tmp/src/github.com/containous/yaegi/interp/cfg.go:1047 +0x19b1
github.com/containous/yaegi/interp.(*Node).Walk(0xc000220780, 0xc0001eb648, 0xc0001eb608)
	/tmp/src/github.com/containous/yaegi/interp/interp.go:101 +0xaf
github.com/containous/yaegi/interp.(*Node).Walk(0xc0002205a0, 0xc0001eb648, 0xc0001eb608)
	/tmp/src/github.com/containous/yaegi/interp/interp.go:98 +0x70
github.com/containous/yaegi/interp.(*Node).Walk(0xc0002204b0, 0xc0001eb648, 0xc0001eb608)
	/tmp/src/github.com/containous/yaegi/interp/interp.go:98 +0x70
github.com/containous/yaegi/interp.(*Node).Walk(0xc000220000, 0xc0001eb648, 0xc0001eb608)
	/tmp/src/github.com/containous/yaegi/interp/interp.go:98 +0x70
github.com/containous/yaegi/interp.(*Interpreter).Cfg(0xc0000cd960, 0xc000220000, 0xc0000ba7cc, 0x2, 0x0, 0x0, 0x1)
	/tmp/src/github.com/containous/yaegi/interp/cfg.go:33 +0x1a9
github.com/containous/yaegi/interp.(*Interpreter).importSrcFile(0xc0000cd960, 0x191bb55, 0x1, 0xc0000ba64b, 0x4, 0xc0000ba64d, 0x2, 0x19000000018ef9a0, 0xc0001eb901)
	/tmp/src/github.com/containous/yaegi/interp/src.go:80 +0x3ee
github.com/containous/yaegi/interp.(*Interpreter).Gta.func1(0xc0000ec3c0, 0x203000)
	/tmp/src/github.com/containous/yaegi/interp/gta.go:135 +0x5fc
github.com/containous/yaegi/interp.(*Node).Walk(0xc0000ec3c0, 0xc0001ebd38, 0x0)
	/tmp/src/github.com/containous/yaegi/interp/interp.go:94 +0xc9
github.com/containous/yaegi/interp.(*Node).Walk(0xc0000ec2d0, 0xc0001ebd38, 0x0)
	/tmp/src/github.com/containous/yaegi/interp/interp.go:98 +0x70
github.com/containous/yaegi/interp.(*Node).Walk(0xc0000ec0f0, 0xc0001ebd38, 0x0)
	/tmp/src/github.com/containous/yaegi/interp/interp.go:98 +0x70
github.com/containous/yaegi/interp.(*Interpreter).Gta(0xc0000cd960, 0xc0000ec0f0, 0xc0000ba640, 0x4, 0x10, 0xc0000ba640)
	/tmp/src/github.com/containous/yaegi/interp/gta.go:26 +0x127
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc0000cd960, 0xc000152070, 0x65, 0x191ccf0, 0x4, 0xc0000cd960, 0x0, 0x1d69080)
	/tmp/src/github.com/containous/yaegi/interp/interp.go:206 +0x104
main.main()
	/tmp/src/github.com/containous/yaegi/yaegi.go:50 +0x404
```

--- a/_test/import3.go
+++ b/_test/import3.go
@@ -2,5 +2,7 @@ package main
 
 import "./foo"
 
-//func main() { println(foo.Bar, foo.Boo) }
 func main() { println(foo.Bar, foo.Boo) }
+
+// Output:
+// BARR Boo

--- a/_test/import4.go
+++ b/_test/import4.go
@@ -1,0 +1,8 @@
+package main
+
+import "./p1"
+
+func main() { println("num:", p1.Uint32()) }
+
+// Output:
+// num: 2596996162

--- a/_test/p1/s1.go
+++ b/_test/p1/s1.go
@@ -1,0 +1,5 @@
+package p1
+
+import "crypto/rand"
+
+var Prime = rand.Prime

--- a/_test/p1/s2.go
+++ b/_test/p1/s2.go
@@ -1,0 +1,7 @@
+package p1
+
+import "math/rand"
+
+var Uint32 = rand.Uint32
+
+func init() { rand.Seed(1) }

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -2,6 +2,7 @@ package interp
 
 import (
 	"fmt"
+	"path"
 	"reflect"
 	"unicode"
 )
@@ -244,7 +245,23 @@ func (interp *Interpreter) Cfg(root *Node) ([]*Node, error) {
 			scope = scope.pushBloc()
 			loop = n
 
-		case ImportSpec, TypeSpec:
+		case ImportSpec:
+			var name, ipath string
+			if len(n.child) == 2 {
+				ipath = n.child[1].val.(string)
+				name = n.child[0].ident
+			} else {
+				ipath = n.child[0].val.(string)
+				name = path.Base(ipath)
+			}
+			if interp.binValue[ipath] != nil && name != "." {
+				scope.sym[name] = &Symbol{kind: Package, typ: &Type{cat: BinPkgT}, path: ipath}
+			} else {
+				scope.sym[name] = &Symbol{kind: Package, typ: &Type{cat: SrcPkgT}, path: ipath}
+			}
+			return false
+
+		case TypeSpec:
 			// processing already done in GTA pass
 			return false
 
@@ -1026,7 +1043,7 @@ func (interp *Interpreter) Cfg(root *Node) ([]*Node, error) {
 			} else {
 				err = n.cfgError("undefined selector: %s", n.child[1].ident)
 			}
-			if n.findex != -1 {
+			if err == nil && n.findex != -1 {
 				n.findex = scope.add(n.typ)
 			}
 

--- a/interp/interp_consistent_test.go
+++ b/interp/interp_consistent_test.go
@@ -171,6 +171,7 @@ func TestInterpErrorConsistency(t *testing.T) {
 			}
 
 			i := interp.New()
+			i.Name = filePath
 			i.Use(stdlib.Value)
 
 			_, errEval := i.Eval(string(src))
@@ -178,7 +179,7 @@ func TestInterpErrorConsistency(t *testing.T) {
 				t.Fatal("An error is expected but got none.")
 			}
 
-			if errEval.Error() != test.expectedInterp {
+			if !strings.Contains(errEval.Error(), test.expectedInterp) {
 				t.Errorf("got %q, want: %q", errEval.Error(), test.expectedInterp)
 			}
 

--- a/interp/interp_eval_test.go
+++ b/interp/interp_eval_test.go
@@ -5,6 +5,7 @@ import (
 	"log"
 	"net/http"
 	"reflect"
+	"strings"
 	"testing"
 	"time"
 
@@ -387,7 +388,7 @@ func assertEval(t *testing.T, i *interp.Interpreter, src, expectedError, expecte
 	res, err := i.Eval(src)
 
 	if expectedError != "" {
-		if err == nil || err.Error() != expectedError {
+		if err == nil || !strings.Contains(err.Error(), expectedError) {
 			t.Fatalf("got %v, want %s", err, expectedError)
 		}
 		return

--- a/interp/interp_file_test.go
+++ b/interp/interp_file_test.go
@@ -52,6 +52,7 @@ func runCheck(t *testing.T, p string) {
 	os.Stdout = w
 
 	i := interp.New()
+	i.Name = p
 	i.Use(interp.ExportValue)
 	i.Use(stdlib.Value)
 
@@ -60,7 +61,7 @@ func runCheck(t *testing.T, p string) {
 		if err == nil {
 			t.Fatalf("got nil error, want: %q", wanted)
 		}
-		if res := strings.TrimSpace(err.Error()); res != wanted {
+		if res := strings.TrimSpace(err.Error()); !strings.Contains(res, wanted) {
 			t.Errorf("got %q, want: %q", res, wanted)
 		}
 		return

